### PR TITLE
Fix unit tests so that we expect to fail 1/5000 times

### PR DESCRIPTION
**Pull request by etwest**
Also some small changes for statistical testing to print out the status

### CODE REVIEW

File: src/graph.cpp [hunk 1]
```diff
@@ -211,15 +211,28 @@ vector<set<node_t>> Graph::parallel_connected_components() {
...
+    #pragma omp parallel for default(none) shared(query, reps, except, err)
     for (node_t i = 0; i < reps.size(); ++i) {
-      auto edge = supernodes[reps[i]]->sample();
+      // wrap in a try/catch because exiting through exception is undefined behavior in OMP
+      boost::optional<Edge> edge;
+      try {
+        edge = supernodes[reps[i]]->sample();
+      } catch (...) {
+        except = true;
+        err = std::current_exception();
```
Comment (Victor-C-Zhang) [inline src/graph.cpp:224]: something to think about:
is there a case where not terminating the running thread would cause weird undefined behavior?

**Comment by Victor-C-Zhang** [comment]
Allowing 2 failures per test for every test seems a bit much. Can we decrease our false positive error rate without ballooning our false negative rate too?

File: src/graph.cpp (hunk 1 above)
Comment (etwest) [reply src/graph.cpp:224]: Yeah it might but the idea is that the code will throw an exception any way right after the parallel loop at that point.

**Comment by etwest** [comment]
> Allowing 2 failures per test for every test seems a bit much. Can we decrease our false positive error rate without ballooning our false negative rate too?

This is entirely possible but our unit test should be quickly alerting us to “this code is very wrong” whereas the statistical testing is responsible for a more fine grained analysis.

Basically this means that a false positive is much worse in this context than a false negative. With say 1 allowed failure per test we end up failing 13 times in 1000 tests.

**Comment by Victor-C-Zhang** [comment]
> > Allowing 2 failures per test for every test seems a bit much. Can we decrease our false positive error rate without ballooning our false negative rate too?
> 
> This is entirely possible but our unit test should be quickly alerting us to “this code is very wrong” whereas the statistical testing is responsible for a more fine grained analysis.
> 
> Basically this means that a false positive is much worse in this context than a false negative. With say 1 allowed failure per test we end up failing 13 times in 1000 tests.

Ok. That makes sense. Since you're doing this 2/10 tolerance a bunch of times, is there a way to wrap this all in a function? If not, can you add a little blurb at the top of the file describing the above rationale?

File: test/parallel_graph_test.cpp [hunk 2]
```diff
@@ -3,6 +3,16 @@
 #include "../include/graph.h"
 #include "util/graph_gen.h"
 
+/**
+ * For many of these tests (especially for those upon very sparse and small graphs)
+ * we allow for a certain number of failures per test.
+ * This is because the responsibility of these tests is to quickly alert us 
+ * to “this code is very wrong” whereas the statistical testing is responsible 
+ * for a more fine grained analysis.
+ * In this context a false false positive is much worse than a false negative.
```
Comment (Victor-C-Zhang) [inline test/parallel_graph_test.cpp]: “false false”

File: test/graph_test.cpp [hunk 3]
```diff
@@ -4,6 +4,16 @@
 #include "util/graph_verifier.h"
 #include "util/graph_gen.h"
 
+/**
+ * For many of these tests (especially for those upon very sparse and small graphs)
+ * we allow for a certain number of failures per test.
+ * This is because the responsibility of these tests is to quickly alert us 
+ * to “this code is very wrong” whereas the statistical testing is responsible 
+ * for a more fine grained analysis.
+ * In this context a false false positive is much worse than a false negative.
```
Comment (Victor-C-Zhang) [inline test/graph_test.cpp]: “false false”

**Review by Victor-C-Zhang — APPROVED** [review]
Approved, delta small typo fix

**Pull request merged**